components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java

```diff
@@ -640,6 +640,15 @@ void setEnvironmentConfig(OMElement environmentElem) throws APIManagementExcepti
             environment.setDescription("");
         }
         environment.setReadOnly(true);
+        OMElement dataPlaneId =
+                environmentElem.getFirstChildWithName(new QName(APIConstants.API_GATEWAY_DATA_PLANE_ID));
+        if (dataPlaneId != null && dataPlaneId.getText() != null && !"".equals(dataPlaneId.getText())) {
+            environment.setDataPlaneId(APIUtil.replaceSystemProperty((dataPlaneId.getText())));
```

Comment:
I think we don't need to use APIUtil.replaceSystemProperty. Nevertheless, from Choreo we don't really set System properties, rather we are setting Environment properties. We can just set the plain value here. From the config management level `$env{ }` placeholders are detected and replaced with environment values automatically. 
eg: https://github.com/wso2-enterprise/choreo-cp-env-overlay/blob/dev/kustomize/dev-control-plane/component-overlay/choreo-apim/config/am/deployment.toml#L111

Reply:
Fixed